src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -254,11 +253,11 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ra
             if (minGeneratedSpan is not null && maxGeneratedSpan is not null)
             {
                 var csharpSourceText = codeDocument.GetCSharpSourceText();
-                var startRange = minGeneratedSpan.Value.AsTextSpan().AsRange(csharpSourceText);
-                var endRange = maxGeneratedSpan.Value.AsTextSpan().AsRange(csharpSourceText);
+                var startRange = minGeneratedSpan.Value.AsTextSpan().AsVSRange(csharpSourceText);
```

Comment:
Would it be rasonable to go directly to VSRange instead of TextSpan -> VSRange?

Reply:
Easier than I thought!